src/elements/modalCreatePool/ModalCreatePool.tsx

```diff
@@ -24,18 +28,79 @@ export const ModalCreatePool = () => {
   const pools = useAppSelector<Pool[]>((state) => state.pool.pools);
   const dispatch = useDispatch();
 
+  const showNotification = (notification: BaseNotification) =>
+    dispatch(addNotification(notification));
+
   const confirm = async () => {
     //isCreateDisabled() TS fails
     if (!account || !chainId || !token) return;
 
-    const notification = await createPool(
+    await createPool(
       token,
       (Number(fee) / 100).toString(),
       chainId,
-      account,
-      (notification: any) => dispatch(addNotification(notification))
+      showNotification({
```

Comment:
I have comments on readability:
1. This becomes a very large file. A lot of this logic can be extracted. specifically the callbacks to show notifications
2. The inline style of the callbacks is hard to read - please consider creating methods with descriptive names

Also, placing this within a function component just leads to redeclaring it. I'd suggest useCallback but I think it just shouldn't be here - these are pure helper functions that receive hash and provide messages